### DISCUSSION TITLE
Enum does not work with Custom Types

### DISCUSSION THREAD

**Discussion by maphe** [mikro-orm/mikro-orm#1129] [Q&A] [answered]
**Describe the bug**
A field that is set as an enum that have a custom type attached to it does not work.

**To Reproduce**
I created a spec here: https://gist.github.com/maphe/01c31121b0f76fdd19c1e4ef16748bb5

Beside the spec failing, SQL log indicates:
```sql
insert into `something` (`directions`) values ('up,down')
```
whereas it should be
```sql
insert into `something` (`directions`) values ('up+down')
```

**Additional context**
I haven't tried with an enum that's not an array cause I don't have the use case, but that'd be worth checking as well.

**Versions**

| Dependency | Version |
| - | - |
| node | 12 |
| typescript | 4.1.2 |
| mikro-orm | 4.3.2 |

**Comment by B4nan** [comment, marked as answer]
Enum array is already a custom type:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/types/EnumArrayType.ts

It is used when you set the `array: true`, and when no custom type is specified via `customType` (not `type`). So just dropping the `array: true` should disable that and prefer your own custom type.

**Comment by maphe** [comment]
Thanks. This is looking better in my app; the test spec is still failing with `DriverException: insert into `something` (`directions`) values ('up+down') - SQLITE_CONSTRAINT: CHECK constraint failed: something` but I don't think it's related to the enum.

Just curious, without the `array: true`, is there any point keeping `items: () => AllowedValues` in the Enum params?

**Comment by B4nan** [comment]
Not really, you don't even have to use the `@Enum` decorator, regular `@Property` is the same here.

**Comment by maphe** [comment]
Fair enough. Thanks.
